src/main/java/uk/gov/cslearning/catalogue/dto/factory/EventDtoFactory.java

```diff
@@ -22,11 +24,25 @@ public EventDto create(Event event, FaceToFaceModule module, Course course) {
         EventDto eventDto = new EventDto();
         eventDto.setId(event.getId());
         eventDto.setModule(moduleDtoFactory.create(module, course));
+        eventDto.setLocation(event.getVenue().getLocation());
+        eventDto.setEventDate(getEventDatesFromDateRanges(event.getDateRanges()));
 
         Optional.ofNullable(course.getLearningProvider())
                 .ifPresent(learningProvider ->
                         eventDto.setLearningProvider(learningProviderDtoFactory.create(learningProvider)));
 
         return eventDto;
     }
-}
+
+    public String getEventDatesFromDateRanges(List<DateRange> dateRangesList) {
+        String eventDate = "";
+        for (int i = 0; i < dateRangesList.size(); i++) {
```

Comment:
Java 8 foreach ???

Reply:
I used for loop here as I'm using the index to add comma between the event dates. as shown here: 
   ```
      if (i != dateRangesList.size() - 1) {
                eventDate += ", ";
            }
```
I'm not sure how I can keep reference for the index using foreach loop

Reply:
public String getEventDatesFromDateRanges(List<DateRange> dateRangesList) {
        return dateRangesList
            .stream()
            .map(date -> date.getDate().toString())
            .collect(Collectors.joining(","));
    }